### PR TITLE
feat(contract): implement execute_transaction function

### LINKED ISSUE #50: [CONTRACT] Implement execute_transaction function in the AccountData Component
## Description 📝
We need to implement a private execute_transaction function that will be called internally to execute approved transactions in our multisignature account system. This function will handle the final execution step after a transaction has received sufficient approvals from designated approvers.
The function needs to:
- Validate that the caller has executor permission
- Verify that the required approval threshold has been met
- Update the transaction status to "executed"
- Update the transaction fields (date_executed and executor)
- Emit appropriate events for transparency
This is a critical component of our multisignature flow as it ensures transactions are only executed when properly authorized and maintains a complete audit trail of execution activities.
## Proposed Steps 📝
- Create the private execute_transaction function with appropriate parameters (transaction ID, caller address)
- Validate that transaction exists.
- Implement permission check to validate the caller has executor rights
- Add logic to verify the number of approvals meets or exceeds the required threshold
- Update the transaction status to "executed" when conditions are met
- Set the date_executed field to the current block timestamp
- Set the executor field to the caller's address
- Emit a TransactionExecuted event with relevant details
- Implement proper error handling for various failure scenarios
## Acceptance Criteria ✅
- [ ] The execute_transaction function is implemented as a private function
- [ ] Function verifies the caller has executor permission before proceeding
- [ ] Function checks that the number of approvers in the approved array meets or exceeds the threshold
- [ ] Transaction status is updated to "executed" upon successful execution
- [ ] The date_executed field is set to the current block timestamp
- [ ] The executor field is set to the caller's address
- [ ] A TransactionExecuted event is emitted with transaction details
- [ ] Appropriate error messages are returned for all failure scenarios
- [ ] Comprehensive unit tests verify all execution scenarios and edge cases
## Additional Context 📌
This function is a core part of our multisignature implementation. The function will only be called internally, likely from a public function that handles the initial validation and passes control to this private function for the actual execution logic. This change will be implemented in the `account_data.cairo` file.

### PR DESCRIPTION
**Pull request by augustin-v**
## Description 📝
This PR implements the private `execute_transaction` function for our multisignature account system as specified in the requirements. The function handles the execution of approved transactions after they've received sufficient approvals from designated approvers.

## Related Issues 🔗
Fixes #50

## Changes Made 🚀
- [x] ✨ Feature Implementation 
- [ ] 🐛 Bug Fix 
- [ ] 📚 Documentation Update 
- [ ] 🔨 Refactoring 
- [ ] ❓ Others (Specify) 

## Checklist ✅
- [x] 🛠 I have tested these changes. 
- [x] 📖 I have updated the documentation (if applicable). 
- [x] 🎨 This PR follows the project's coding style. 
- [x] 🧪 I have added necessary tests (if applicable). 

## Additional Notes 🗒
The implementation includes:
- A private `execute_transaction` function that validates executor permissions, checks approval threshold, updates transaction status, and records execution details
- A new `TransactionExecuted` event for transparency
- Comprehensive tests for successful execution and all error cases
- All tests are passing, including validation of transaction status updates and proper timestamp handling

### CODE REVIEW

**Comment by codebestia** [comment]
@augustin-v Well done! Your implementation looks correct and follows the expected approach. Kindly resolve the conflicts so we can merge this.

File: src/tests/test_account_data.cairo [hunk 1]
```diff
@@ -501,6 +504,11 @@ fn test_cannot_approve_transaction_more_than_once() {
 
     stop_cheat_caller_address(mock_contract.contract_address);
 }
+#[test]
+fn test_execute_transaction_successful() {
+    let mock_contract = deploy_mock_contract();
+    let caller = member();
+}
```
Comment (codebestia) [inline src/tests/test_account_data.cairo:511]: Is this the only test for the successful transaction execution?

File: src/tests/test_account_data.cairo [hunk 2]
```diff
@@ -321,6 +321,9 @@ fn test_approve_transaction_successful() {
     stop_cheat_caller_address(mock_contract.contract_address);
 
     let transaction = mock_contract.get_transaction_pub(tx_id);
+    println!("transaction date: {}", transaction.date_executed);
+    println!("block timestamp: {}", starknet::get_block_timestamp());
+
```
Comment (codebestia) [inline src/tests/test_account_data.cairo:326]: Please remove the debug println function.

**Review by codebestia — CHANGES_REQUESTED** [review]
Kindly address the minor changes requested. Good Job so far

File: src/tests/test_account_data.cairo (hunk 1 above)
Comment (augustin-v) [reply src/tests/test_account_data.cairo]: Sorry for the overlook seems like it got cut off when rebasing, fixed it

File: src/tests/test_account_data.cairo (hunk 2 above)
Comment (augustin-v) [reply src/tests/test_account_data.cairo]: Done

**Comment by augustin-v** [comment]
Thanks for the review again @codebestia! I addressed the changes looking forward to the review

**Review by codebestia — APPROVED** [review]
LGTM!
Thank you for your contribution @augustin-v
